src/proto-libs/internal.h

```diff
@@ -0,0 +1,43 @@
+#ifndef MOTORS_INTERNAL_H
+#define MOTORS_INTERNAL_H
+
+//TODO: do we need this import?
+#include "api.h"
+#include "stdint.h"
+
+typedef enum {
+    BACKWARD = 0,
+    FORWARD = 1
+} Direction;
+
+/* Contains mutable state for single motor */
+typedef struct {
+    Direction direction;
+    int current_steps;
+    int max_steps;
+    int min_steps;
+    uint32_t step_delay_us;
+    uint32_t last_step_time_us;
+
+    // TODO: we should reguaraly update these fields if they are present here.
+    /* bool max_triggered; */
+    /* bool min_triggered; */
+} MotorState;
+
+
+/* Contains attributes of a single motor.
+* Filled at the initialization phase.
+*/
+struct MotorAttributes {
+    MovementResult (* proto_specific_move)(MotorState* state, int steps);
```

Comment:
Typedef?

Reply:
fixed in [83a4b31](https://github.com/ThatDraenGuy/motorland/pull/5/commits/83a4b31c987b41c04cb391f9750f26b95757db58)